src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -1,22 +1,63 @@
 package ma.hibernate.dao;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
 import ma.hibernate.model.Phone;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
 
 public class PhoneDaoImpl extends AbstractDao implements PhoneDao {
     public PhoneDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
-
+    
```

Comment:
```suggestion

```

Reply:
I didn't find extra spaces in my project..